verilog/tools/kythe/indexing_facts_tree_extractor.cc

```diff
@@ -1458,6 +1476,23 @@ void IndexingFactsTreeExtractor::ExtractTypeDeclaration(
   }
 }
 
+void IndexingFactsTreeExtractor::ExtractTemporaryScope(
+    const verible::SyntaxTreeNode& node) {
+  IndexingFactNode temp_scope_node(
+      IndexingNodeData{IndexingFactType::kTemporaryScope});
+
+  // Generate unique id for this scope.
+  temp_scope_node.Value().AppendAnchor(
+      Anchor(absl::StrCat("temp-scope-", getpid(), "-", random()), 0, 0));
```

Comment:
Choosing `-` is good because it cannot appear in any user space identifier.

I would also use one counter **per enclosing scope** instead of `random()` for determinism, and maximum stability (to incremental changes).  
e.g. `int next_anonymous_id = 0;`